builtin/providers/aws/resource_aws_dynamodb_table.go

```diff
@@ -336,6 +366,32 @@ func resourceAwsDynamoDbTableUpdate(d *schema.ResourceData, meta interface{}) er
 		waitForTableToBeActive(d.Id(), meta)
 	}
 
+	if d.HasChange("stream_specification") {
+		log.Printf("[DEBUG] Changed Stream Specification")
+		req := &dynamodb.UpdateTableInput{
+			TableName: aws.String(d.Id()),
+		}
+
+		vs := d.Get("stream_specification").(*schema.Set).List()
+		if len(vs) > 0 {
+			spec := vs[0].(map[string]interface{})
+			streamSpecification := &dynamodb.StreamSpecification{
+				StreamEnabled:   aws.Bool(bool(spec["enabled"].(bool))),
+				StreamViewType: aws.String(string(spec["view_type"].(string))),
+			}
+			req.StreamSpecification = streamSpecification
+			d.SetPartial("stream_specification")
```

Comment:
I think you're doing the right thing here, but I don't see ``SetPartial`` called anywhere else in here and this function also isn't enabling partial state (by calling ``d.Partial(true)``), so I don't think this line is actually doing anything in this case.

The usual way to use partial state is to turn it on at the beginning of the function, then call ``SetPartial`` each time one of the properties is committed, and then finally call ``Partial(false)`` to flush out any remaining changes that haven't yet been explicitly flushed.